src/Sextant.WinUI/Mixins/DependencyResolverMixins.cs

```diff
@@ -0,0 +1,191 @@
+            this IMutableDependencyResolver dependencyResolver, Func<TView> navigationViewFactory)
+            where TView : IView
+        {
+            if (dependencyResolver is null)
+            {
+                throw new ArgumentNullException(nameof(dependencyResolver));
+            }
+
+            if (navigationViewFactory is null)
+            {
+                throw new ArgumentNullException(nameof(navigationViewFactory));
+            }
+
+            Lazy<(TView, ViewStackService)> lazy = new Lazy<(TView, ViewStackService)>(() =>
+            {
+                var navigationView = navigationViewFactory();
+                var viewStackService = new ViewStackService(navigationView);
+
+                return (navigationView, viewStackService);
+            });
+
+            dependencyResolver.RegisterLazySingleton<IViewStackService>(() => lazy.Value.Item2);
+            dependencyResolver.RegisterLazySingleton<IView>(() => lazy.Value.Item1, NavigationView);
+            return dependencyResolver;
+        }
+
+        /// <summary>
+        /// Gets the navigation view.
+        /// </summary>
+        /// <param name="dependencyResolver">The dependency resolver.</param>
+        /// <param name="contract">The contract.</param>
+        /// <returns>The navigation view.</returns>
+        public static NavigationView? GetNavigationView(
+            this IReadonlyDependencyResolver dependencyResolver,
+            string? contract = null) =>
+            dependencyResolver.GetService<IView>(contract ?? NavigationView) as NavigationView;
+
+        /// <summary>
+        /// Initializes WinUI-specific view locator.
+        /// </summary>
+        /// <param name="dependencyResolver">The dependency resolver.</param>
+        /// <returns>The dependencyResolver.</returns>
+        public static IMutableDependencyResolver RegisterWinUIViewLocator(
+            this IMutableDependencyResolver dependencyResolver)
+        {
+            if (dependencyResolver is null)
+            {
+                throw new ArgumentNullException(nameof(dependencyResolver));
+            }
+
+            dependencyResolver.RegisterConstant(new ViewTypeResolver(), typeof(ViewTypeResolver));
+            return dependencyResolver;
+        }
+
+        /// <summary>
+        /// Register view for viewmodel, but only return view type for WinUI frame.
+        /// </summary>
+        /// <typeparam name="TView">The view type.</typeparam>
+        /// <typeparam name="TViewModel">The viewmodel type.</typeparam>
+        /// <param name="dependencyResolver">The dependency resolver.</param>
+        /// <param name="contract">The contract.</param>
+        /// <returns>
+        /// The dependencyResolver.
+        /// </returns>
+        public static IMutableDependencyResolver RegisterViewWinUI<TView, TViewModel>(
+            this IMutableDependencyResolver dependencyResolver, string? contract = null)
+            where TView : IViewFor<TViewModel>
```

Comment:
The UWP-version has a `new()` constraint here.
Based on the description, shouldn't DI be used to retrieve the instance?
NOTE: I changed this (see line 143).